packages/hydrogen-react/src/RichText.tsx

```diff
@@ -0,0 +1,135 @@
+import {createElement, type ReactNode} from 'react';
+
+export interface RichTextASTNode {
+  type: string;
```

Comment:
Let's add an enum with all the possible type 🙏🏼

Reply:
Not enum - @frehner can give you the download on why

Here is an example of what I did to get constant types and types for documentation generation
https://github.com/Shopify/hydrogen/blob/2023-01/packages/hydrogen-react/src/analytics-constants.ts